apps/builder/app/canvas/features/webstudio-component/link.ts

```diff
@@ -0,0 +1,53 @@
+import type { MouseEvent } from "react";
+import { useStore } from "@nanostores/react";
+import { findPageByIdOrPath, Page } from "@webstudio-is/project-build";
+import { pagesStore, useIsPreviewMode } from "~/shared/nano-states";
+import { publish } from "~/shared/pubsub";
+
+declare module "~/shared/pubsub" {
+  export interface PubsubMap {
+    switchPage: { pageId: Page["id"] | "home" };
+  }
+}
+
+const isAbsoluteUrl = (href: string) => {
+  try {
+    new URL(href);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const useHandleLinkClick = () => {
+  const pages = useStore(pagesStore);
+  const [isPreviewMode] = useIsPreviewMode();
+
+  return (event: MouseEvent) => {
+    event.preventDefault();
+
+    if (isPreviewMode === false || pages === undefined) {
+      return;
+    }
+
+    const href = event.currentTarget.getAttribute("href");
+
+    if (href === null) {
+      return;
+    }
+
+    const page = findPageByIdOrPath(pages, href);
+
+    if (page) {
+      publish({
+        type: "switchPage",
+        payload: { pageId: page.id === pages.homePage.id ? "home" : page.id },
```

Comment:
homePage have actual id why use "home"?

Reply:
This is just because we want to set remove `?pageId` from URL when it's home. So we need access to `pages.homePage.id` to check whether this is the home page. I decided to check here since I already have `pages.homePage` here.

Reply:
It's just confusing that pageId can be not actual page id.

Reply:
Removed "home"